extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSS} from '../../../build/amp-story-quiz-1.0.css';
+import {createShadowRootWithStyle} from '../0.1/utils';
+import {htmlFor} from '../../../src/static-template';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {setStyle} from '../../../src/style';
+
+export class AmpStoryQuiz extends AMP.BaseElement {
+  /**
+   * @param {!AmpElement} element
+   */
+  constructor(element) {
+    super(element);
+
+    /** @private {?ShadowRoot} */
+    this.shadowRoot_ = null;
+
+    /** @private {boolean} */
+    this.hasReceivedResponse_ = false;
+
+    /** @private {Array<number>} */
+    this.percentages_ = this.TEMPgenerateRandomPercentages_();
+
+    /** @private {Array<string>} */
+    this.answerChoiceOptions_ = ['A', 'B', 'C', 'D'];
+
+    this.configureOption_ = this.configureOption_.bind(this);
+  }
+
+  /** @override */
+  buildCallback() {
+    const html = htmlFor(this.element);
+    const shadowElement = html`
```

Comment:
Can you move this to a file-level constant?

Reply:
I don't think I can, the htmlFor method requires the element for each AmpStoryQuiz, and es-lint doesn't allow me to do something like this:

```js
const templateString = `...`;
...
const shadowElement = html`${templateString}`;
```

`The html template tag CANNOT accept expression. The template MUST be static only.eslint(local/html-template)`

Reply:
Can you try writing it like this maybe?

https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/1.0/amp-story-page.js#L136